core/Url.php

```diff
@@ -669,6 +664,24 @@ public static function isHostInUrls($host, $urls)
             }
         }
 
-        return in_array($host, self::$alwaysTrustedHosts);
+        return in_array($host, self::getAlwaysTrustedHosts());
+    }
+
+    /**
+     * List of hosts that are never checked for validity.
+     *
+     * @return array
+     */
+    private static function getAlwaysTrustedHosts()
+    {
+        return self::getLocalHostnames();
+    }
+
+    /**
+     * @return array
+     */
+    public static function getLocalHostnames()
+    {
+        return array('localhost', '127.0.0.1', '::1', '[::1]');
```

Comment:
And what about using `$_SERVER['SERVER_ADDR']` and `gethostbyname(gethostname())`?